test/embind/test_return_value_policy.cpp

```diff
@@ -264,7 +331,27 @@ EMSCRIPTEN_BINDINGS(xxx) {
     .constructor<>()
     .function("getPtrTakeOwnership", &ValueHolder::getNewPtr, return_value_policy::take_ownership())
     .function("getPtrReference", &ValueHolder::getPtr, return_value_policy::reference())
-    .function("getPtrWithStdFunctionTakeOwnership", std::function<Value*(ValueHolder&)>(&value_holder), return_value_policy::take_ownership())
-    .class_function("staticGetPtrReference", &ValueHolder::staticGetPtr, return_value_policy::reference());
+    .function("getPtrWithStdFunctionTakeOwnership", std::function<const Value*(const ValueHolder&)>(&value_holder), return_value_policy::take_ownership())
+    .class_function("staticGetPtrReference", &ValueHolder::staticGetPtr, return_value_policy::reference())
+    .function("initializeValuePtr", &ValueHolder::initializePtr)
+
+    // Test the code path for properties directly bound to a member.
+    .property("valueDefault", &ValueHolder::value)
+    .property("valueReference", &ValueHolder::value, return_value_policy::reference())
+    .property("valueTakeOwnership", &ValueHolder::value, return_value_policy::take_ownership())
+
+    // Test binding raw pointers with the various methods.
+    .property("valuePtrReference", &ValueHolder::valuePtr, return_value_policy::reference())
+    .property("valuePtrTakeOwnership", &ValueHolder::valuePtr, return_value_policy::take_ownership())
+
+    // Check that compiling with getter and setter methods works as expected.
+    // These are only tested for compilation since they use the same underlying
+    // mechanisms as the other property bindings.
+    // Getter only.
+    .property("getterValue", &ValueHolder::get, return_value_policy::reference())
+    // Getter and setter.
+    .property("getterSetterValue", &ValueHolder::get, &ValueHolder::set, return_value_policy::reference())
+    // std::function getter.
+    .property("getterPtrWithStdFunctionReference", std::function<const Value*(const ValueHolder&)>(&value_holder), return_value_policy::reference())
     ;
```

Comment:
I think the use case I had in mind for this feature was with raw pointers - is that supported as well? If so can it be tested here?

Reply:
It's supported and tested above. https://github.com/emscripten-core/emscripten/pull/21935/files#diff-4fe14676abdaa75c05abe1f28bf3cd5c2dbe96d8abc99f7c893fa883b89e06e5R343-R345

Reply:
Thanks, I see.

Though I was expecting to read `allow_raw_pointers()`. Is that not needed in these cases? I can't tell from the docs.

Reply:
I'm hoping to deprecate `allow_raw_pointers()` in favor of return policies. Using a return value policy implies that raw pointers are allowed. There's a little note on this [here](https://emscripten.org/docs/porting/connecting_cpp_and_javascript/embind.html#raw-pointers). I should probably update that to be more generic now that RVP is allowed on properties too.

Reply:
I see, thanks!